src/oc_span_sweeper.erl

```diff
@@ -58,24 +61,10 @@ handle_event(state_timeout, sweep, _, #data{interval=Interval,
         NumDeleted ->
             ?LOG_INFO("sweep old spans: ttl=~p num_dropped=~p", [TTL, NumDeleted])
     end,
-    {keep_state_and_data, [hibernate, {state_timeout, Interval, sweep}]};
-handle_event(state_timeout, sweep, _, #data{interval=Interval,
-                                            strategy=finish,
-                                            ttl=TTL}) ->
-    Expired = select_expired(TTL),
-    [finish_span(Span) || Span <- Expired],
-    {keep_state_and_data, [hibernate, {state_timeout, Interval, sweep}]};
-handle_event(state_timeout, sweep, _, #data{interval=Interval,
-                                            strategy=failed_attribute_and_finish,
-                                            ttl=TTL}) ->
-    Expired = select_expired(TTL),
-    [finish_span(oc_span:put_attribute(<<"finished_by_sweeper">>, true, Span)) || Span <- Expired],
-    {keep_state_and_data, [hibernate, {state_timeout, Interval, sweep}]};
-handle_event(state_timeout, sweep, _, #data{interval=Interval,
-                                            strategy=Fun,
-                                            ttl=TTL}) when is_function(Fun) ->
-    Expired = select_expired(TTL),
-    [Fun(Span) || Span <- Expired],
+    {keep_state_and_data, [{next_event, internal, just_do_it},
```

Comment:
What is this new internal event?

Reply:
copy-pasta, at the beginning I wanted to first sweep by ttl and then memory, that's a left over